src/polynomial/grouped_poly.rs

```diff
@@ -68,51 +84,78 @@ impl<F: PrimeField> GroupedPoly<F> {
     ///     a2*d1 + a2*e1 + b2*d1 + b2*e1 + c2*d1 + c2*e1 + a1*d2 + a1*e2 + b1*d2 + b1*e2 + c1*d2 + c1*e2 * k^1 +
     ///     a2*d2 + a2*e2 + b2*d2 + b2*e2 + c2*d2 + c2*e2       * k^2
     /// ```
-    pub fn new(
-        expr: &Expression<F>,
-        num_selectors: usize,
-        num_fixed: usize,
-        num_of_fold_vars: usize,
-        num_challenges: usize,
-    ) -> Self {
-        expr.evaluate(
-            &|constant| GroupedPoly {
-                terms: vec![Some(Expression::Constant(constant))],
+    pub fn new(expr: &Expression<F>, ctx: &QueryIndexContext) -> Self {
+        use Expression::*;
```

Comment:
In case I miss it, besides QueryType and "closure -> match". Is there any other important change in this function?

Reply:
Not, that's it